api/chat_main_handler.go

```diff
@@ -285,23 +227,12 @@ func (h *ChatHandler) OpenAIChatCompletionAPIWithStreamHandler(w http.ResponseWr
 	} else {
 
 		// Set up SSE headers
-		answerText, answerID, shouldReturn := chat_stream(ctx, chatSession, msgs, w)
+		answerText, answerID, shouldReturn := chat_stream(w, chatSession, msgs)
 		if shouldReturn {
 			return
 		}
 		// insert ChatMessage into database
-		chatMessage := sqlc_queries.CreateChatMessageParams{
-			Uuid:            answerID,
-			ChatSessionUuid: chatSessionUuid,
-			Role:            "assistant",
-			UserID:          int32(userIDInt),
-			Content:         answerText,
-			CreatedBy:       int32(userIDInt),
-			UpdatedBy:       int32(userIDInt),
-			Raw:             json.RawMessage([]byte("{}")),
-		}
-
-		_, err := h.chatService.q.CreateChatMessage(ctx, chatMessage)
+		_, err := h.chatService.CreateChatMessageSimple(ctx, chatSessionUuid, answerID, "assistant", answerText, userID)
 
 		if err != nil {
 			RespondWithError(w, http.StatusInternalServerError, fmt.Errorf("fail to create message: %w", err).Error(), nil)
```

Comment:
:robot: OpenAI

This patch changes the `OpenAIChatCompletionAPIWithStreamHandler` function in `api/chat_main_handler.go` to use the `CreateChatMessageSimple` method of `h.chatService`.

This new method is not defined in this file, so it must be defined elsewhere. If it's not defined properly, this change could cause issues.

Please provide more context on where `CreateChatMessageSimple` is defined and how it works.

<!-- This is an auto-generated comment by OpenAI -->